src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1059,14 +1062,26 @@ void OrbitMainWindow::on_actionOpen_Capture_triggered() {
     return;
   }
 
-  OpenCapture(file.toStdString());
+  const bool starts_with_file_target = (target_configuration_.index() == 2);
+  const bool has_data = app_->HasCaptureData();
+  // If the current OrbitMainWindow starts with a FileTarget and the capture file is cleared, open
+  // new capture file in the current OrbitMainWindow. Otherwise, open capture files in new windows.
+  OpenCapture(file.toStdString(), !starts_with_file_target || has_data);
 }
 
-void OrbitMainWindow::OpenCapture(const std::string& filepath) {
-  app_->OnLoadCapture(filepath);
-  setWindowTitle(QString::fromStdString(filepath));
-  UpdateCaptureStateDependentWidgets();
-  FindParentTabWidget(ui->CaptureTab)->setCurrentWidget(ui->CaptureTab);
+void OrbitMainWindow::OpenCapture(const std::string& filepath, bool open_in_new_window) {
+  if (open_in_new_window) {
+    QString orbit_executable =
+        QString::fromStdString(orbit_base::GetExecutablePath().generic_string());
+    QStringList arguments;
+    QProcess::startDetached(orbit_executable,
+                            arguments << QString::fromStdString(filepath) << command_line_flags_);
+  } else {
```

Comment:
personal preference: I would put this into `OrbitMainWindow::on_actionOpen_Capture_triggered` (no need for that extra parameter `open_in_new_window`)

Reply:
Nice catch! And by doing so there is no need to change QString (i.e., `file` in `on_actionOpen_Capture_triggered`) to string and then back to QString. 

Changed accordingly. Thanks!